### PR TITLE
Register

### PR DESCRIPTION
**Pull request by Nosoyleon**
## Summary

- Register component
- Some globals styles
- Inicial data manipulation
- Submit data to console

## Screenshots

image

## Trello Card
https://trello.com/c/oTX5X8fs/22-register-layout-exercise

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,32 +1,27 @@
 <template>
   <div id="app">
```
Comment (CrisVega08) [inline src/App.vue]: Use pug
```suggestion
 #app
    
```

File: src/App.vue [hunk 2]
```diff
@@ -1,32 +1,27 @@
...
-    <div id="nav">
-      <router-link to="/">Home</router-link> |
-      <router-link to="/about">About</router-link>
-    </div>
-    <router-view/>
+    <register />
```
Comment (CrisVega08) [inline src/App.vue]: Same here

File: src/App.vue [hunk 3]
```diff
@@ -1,32 +1,27 @@
...
 
+<script>
+import Register from '@/views/Register.vue'
+
+export default {
+  components: {
+    Register
+  }
+}
+</script>
```
Comment (CrisVega08) [inline src/App.vue]: Add empty line

File: src/scss/buttons.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
+@import './variables/colors';
+
+%button {
```
Comment (CrisVega08) [inline src/scss/buttons.scss]: Use a more descriptive name for example %base-button

File: src/scss/buttons.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  cursor: pointer;
+  font-weight: bold;
+  text-align: center;
+  width: 100%;
+}
+
+.primary-button {
+  background-color: $atlantis;
+  color: $white;
+  @extend %button;
```
Comment (CrisVega08) [inline src/scss/buttons.scss]: Use extends at first line of class

File: src/scss/buttons.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+.secundary-button {
+  background-color: $gallery;
+  color: $atlantis;
+  @extend %button;
```
Comment (CrisVega08) [inline src/scss/buttons.scss]: Same here

File: src/scss/variables/_colors.scss [hunk 7]
```diff
@@ -0,0 +1,13 @@
+$atlantis: #A6CB46;
```
Comment (CrisVega08) [inline src/scss/variables/_colors.scss]: Do you use all these colors??

File: src/views/Register.vue [hunk 8]
```diff
@@ -0,0 +1,84 @@
+<template>
+  <div :class="$style.container">
```
Comment (CrisVega08) [inline src/views/Register.vue]: Use pug throughout the file, I don't want to put in each line `same here`
```suggestion
  .container
```

File: src/views/Register.vue [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  }),
+  methods: {
+    onSubmit () {
+      console.log('Submited Data', this.user)
+    }
+  }
+}
+</script>
+
+<style module lang="scss">
```
Comment (CrisVega08) [inline src/views/Register.vue]: add scoped property

File: src/views/Register.vue [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        <button type="submit" :class="[$style.button, 'primary-button']">{{signIn}}</button>
+        <div :class="$style.separator" />
+        <button :class="[$style.button, 'secundary-button']">{{logIn}}</button>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script>
+import '@/scss/buttons.scss'
```
Comment (CrisVega08) [inline src/views/Register.vue]: Add empty line

File: src/views/Register.vue [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+import '@/scss/inputs.scss'
+
+export default {
+  name: 'register',
+  data: () => ({
+    fields: [
```
Comment (CrisVega08) [inline src/views/Register.vue]: This can be a constant on a different file

File: src/views/Register.vue [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  }
+}
+</script>
+
+<style module lang="scss">
+@import '@/scss/variables/_colors';
+
+.container {
+  margin: 0 auto;
```
Comment (CrisVega08) [inline src/views/Register.vue]: Sort properties

File: src/views/Register.vue [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+.button {
+  padding: 10px;
+}
+
+.logo {
+  width: 100%;
+}
+
+.title  {
+  text-transform: uppercase;
```
Comment (CrisVega08) [inline src/views/Register.vue]: Sort

File: src/scss/inputs.scss [hunk 14]
```diff
@@ -0,0 +1,22 @@
+@import './variables/colors';
+
+.input-container {
+  display: flex;
+  flex-direction: column;
+  margin-bottom: 20px;
+}
+
+.label {
```
Comment (CrisVega08) [inline src/scss/inputs.scss]: change name

File: src/scss/inputs.scss [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+  align-self: flex-start;
+  color: $black;
+  font-size: 12px;
+  font-weight: bold;
+  margin-left: 12px;
+}
+
+.input {
```
Comment (CrisVega08) [inline src/scss/inputs.scss]: Same

File: src/views/Register.vue [hunk 16]
```diff
@@ -0,0 +1,84 @@
+<template>
+  <div :class="$style.container">
+    <div :class="$style.header">
```
Comment (CrisVega08) [inline src/views/Register.vue]: You don't need use $style

File: src/views/Register.vue [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+      <img :class="$style.logo" alt="wolox" src="@/assets/logo.png" />
+      <h1 :class="$style.title">{{title}}</h1>
```
Comment (CrisVega08) [inline src/views/Register.vue]: Always separate tags and properties of text to render
```suggestion
      h1.title
        |  {{ title }}
```

File: src/views/Register.vue [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+    </div>
+    <form id="register" @submit.prevent="onSubmit">
+      <div class="input-container" v-for="field of fields" :key="field.id">
```
Comment (CrisVega08) [inline src/views/Register.vue]: Nice the form, too bad you don't use pug

File: src/views/Register.vue [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+.container {
+  margin: 0 auto;
+  align-self: center;
+  max-width: 300px;
+  background-color: $gallery;
+  padding: 0 10px 20px;
+  border-top: 3px solid $cerulean;
+}
+
+.button {
```
Comment (CrisVega08) [inline src/views/Register.vue]: same about names

File: src/views/Register.vue [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+  padding: 10px;
+}
+
+.logo {
+  width: 100%;
+}
+
+.title  {
```
Comment (CrisVega08) [inline src/views/Register.vue]: same about names

File: src/App.vue [hunk 21]
```diff
@@ -1,32 +1,27 @@
...
 <style lang="scss">
 #app {
+  color: #2C3E50;
+  display: flex;
   font-family: 'Avenir', Helvetica, Arial, sans-serif;
   -webkit-font-smoothing: antialiased;
+  height: 100vh;
```
Comment (CrisVega08) [inline src/App.vue]: sort properties

File: src/scss/variables/_colors.scss (hunk 7 above)
Comment (Nosoyleon) [reply src/scss/variables/_colors.scss]: No, I don't

Comment (Nosoyleon) [reply src/scss/variables/_colors.scss]: I am 70% sure that a will need them, at least the most

File: src/views/Register.vue [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+      console.log('Submited Data', this.user)
+    }
+  }
+}
+</script>
+
+<style scoped lang="scss">
+
+@import '@/scss/variables/_colors';
+@import '@/scss/buttons.scss';
```
Comment (CrisVega08) [inline src/views/Register.vue]: Buttons and Inputs are global styles, why do you import these here?

File: src/views/Register.vue [hunk 23]
```diff
@@ -0,0 +1,80 @@
...
+  .container
+    .header
+      img.form-logo(alt="wolox" src="@/assets/logo.png")
+      h1.title
+        | {{title}}
+    form#register(@submit.prevent="onSubmit")
+      .input-container(v-for="field of fields" :key="field.id")
+        label.base-label(:for="field.name")
+          | {{field.label}}
+        input.base-input(:type="field.type" :name="field.name" :id="field.name" v-model="user[field.model]")
```
Comment (CrisVega08) [inline src/views/Register.vue]:
```suggestion
        input.base-input(
          :type="field.type"
          :name="field.name"
          :id="field.name"
          v-model="user[field.model]"
         )
```

File: src/views/Register.vue [hunk 24]
```diff
@@ -0,0 +1,80 @@
...
+      .buttons
+        button.form-button.primary-button(type="submit")
+          | {{signIn}}
```
Comment (CrisVega08) [inline src/views/Register.vue]:
```suggestion
          | {{ signIn }}
```

File: src/views/Register.vue [hunk 25]
```diff
@@ -0,0 +1,80 @@
...
+        .separator
+        button.form-button.secundary-button
+          | {{logIn}}
```
Comment (CrisVega08) [inline src/views/Register.vue]:
```suggestion
          | {{ logIn }}
```

File: src/views/Register.vue [hunk 26]
```diff
@@ -0,0 +1,80 @@
+<template lang="pug">
+  .container
+    .header
+      img.form-logo(alt="wolox" src="@/assets/logo.png")
+      h1.title
+        | {{title}}
+    form#register(@submit.prevent="onSubmit")
+      .input-container(v-for="field of fields" :key="field.id")
+        label.base-label(:for="field.name")
+          | {{field.label}}
```
Comment (CrisVega08) [inline src/views/Register.vue]:
```suggestion
          | {{ field.label }}
```

File: src/views/Register.vue [hunk 27]
```diff
@@ -0,0 +1,80 @@
+<template lang="pug">
+  .container
+    .header
+      img.form-logo(alt="wolox" src="@/assets/logo.png")
+      h1.title
+        | {{title}}
```
Comment (CrisVega08) [inline src/views/Register.vue]:
```suggestion
        | {{ title }}
```

File: src/views/Register.vue [hunk 28]
```diff
@@ -0,0 +1,80 @@
...
+        button.form-button.secundary-button
+          | {{logIn}}
+</template>
+
+<script>
+import { REGISTER_FIELDS } from './constants'
+
+export default {
+  name: 'register',
+  data: () => ({
```
Comment (CrisVega08) [inline src/views/Register.vue]:
```suggestion
  data(): {
  }
```

**Pull request merged**